contracts/infrastructure/storage/Storage.sol

```diff
@@ -17,19 +17,31 @@
 pragma solidity >=0.5.4 <0.7.0;
 
 import "../../wallet/IWallet.sol";
+import "../../modules/common/IFeature.sol";
 
 /**
  * @title Storage
  * @notice Base contract for the storage of a wallet.
- * @author Julien Niset - <julien@argent.xyz>
+ * @author Julien Niset - <julien@argent.xyz>, Olivier VDB - <olivier@argent.xyz>
  */
 contract Storage {
 
     /**
      * @notice Throws if the caller is not an authorised module.
      */
     modifier onlyModule(address _wallet) {
-        require(IWallet(_wallet).authorised(msg.sender), "TS: must be an authorized module to call this method");
+        require(IWallet(_wallet).authorised(msg.sender), "S: must be an authorized module to call this method");
+        _;
+    }
+
+    /**
+     * @notice Throws if the caller is not an authorised feature.
+     */
+    modifier onlyFeature(address _wallet) {
+        require(
+            IFeature(msg.sender).isFeatureAuthorisedInVersionManager(_wallet, msg.sender), 
+            "S: must be an authorized feature to call this method"
+        );
```

Comment:
This is not ok since we do the check with msg.sender itself. Anybody can modify the storage by creating a custom contract with a `isFeatureAuthorisedInVersionManager` method that always returns `true` .

Reply:
Unfortunately I think the only solution is to only let modules modify storages and have features proxy through the `VersionManager`. This will make the model cleaner by treating all the storages (old and new) coherently, but it will add an extra `call` (~2000 gas) to every write.